binary.js

```diff
@@ -1,30 +1,81 @@
 'use strict'
+/* 
+PSEUDOCODE
 
-var testArrayGenap = [40, 18, 22, 32, 90, 10, 10, 22, 8]
-var testArrayGanjil = [3, 31, 89, 53, 53, 85, 77, 21, 55]
+DEF ownSort with parameter arr
+  FOR i = 0 and i < length of array, i increment by 1, THEN
+    FOR j = i and j>= 0, j decrease by 1, THEN
+      IF value of arr with index of next number of j < value of arr with index of j, THEN
+        SET temp with value of arr with index j
+        SET value of arr with index j to value of arr with index of next number of j
+        SET value of arr with index of next number of j with temp
+      ENDIF
+    ENDFOR
+  ENDFOR
+  RETURN arr as result
+
+DEF binarySearch with parameter search and array
+  SET high as number with value length of array minus 1
+  SET low as number with value 0
+  WHILE low less than or equal to high,THEN
+    SET mid as number with value sum of low and high and divided by two
+    mid will be rounded to smallest decimal place
+    IF value of array with index of mid equal to search,THEN
+      RETURN mid as result
+    ELSE IF search < value of array with index of mid,THEN
+      high equal mid - 1;
+    ELSE
+      low = mid + 1;
+    ENDIF
+  ENDLOOP
+  RETURN -1 as result
+*/
+
+var testArrayGenap = [40, 18, 22, 32, 90, 10, 10, 22, 8];
+var testArrayGanjil = [3, 31, 89, 53, 53, 85, 77, 21, 55];
 
 function ownSort(arr) {
-  // Your sorting code
+  for (var i = 0; i<arr.length;i++){
+    for (var j = i ; j>=0;j--){
+      if (arr[j+1]<arr[j]){
+        var temp = arr[j];
+        arr[j] = arr[j+1];
+        arr[j+1] =temp;
+      }
+    }
+  }
   return arr
 }
 
 function binarySearch (search, array) {
-  // Your searching code
-  return 0;
+  var high = array.length-1;
+  var low = 0;
+  while(low<=high){
+    var mid = Math.floor((low+high)/2);
+    if(array[mid]==search){
```

Comment:
Penulisannya harus dirapihkan lagi yaa

```suggestion
    if(array[mid] == search){
```